internal/db/repos/instance.go

```diff
@@ -244,7 +244,8 @@ func (r *InstanceRepository) GetByJobID(ctx context.Context, ownerID, jobID uint
 	var instances []models.Instance
 	query := r.db.WithContext(ctx).
 		Unscoped().
-		Where(&models.Instance{JobID: jobID})
+		Where(&models.Instance{JobID: jobID}).
+		Where("status != ?", models.InstanceStatusTerminated)
```

Comment:
[non blocking] follow up here, we should just extend the opts to this method so that we can filter by status and not just this one status.